src/java.base/share/classes/java/io/BufferedReader.java

```diff
@@ -394,12 +398,74 @@ String readLine(boolean ignoreLF, boolean[] term) throws IOException {
      *
      * @throws     IOException  If an I/O error occurs
      *
+     * @see #readAllLines
      * @see java.nio.file.Files#readAllLines
      */
     public String readLine() throws IOException {
         return readLine(false, null);
     }
 
+    /**
+     * Reads all remaining lines of text. A line is considered to be terminated
+     * by any one of a line feed ('\n'), a carriage return ('\r'), a carriage
+     * return followed immediately by a line feed, or by reaching the
+     * end-of-file (EOF).
+     *
+     * <p> This method works as if invoking it were equivalent to evaluating
+     * the expression:
+     * <blockquote>{@link #lines()}.toList()</blockquote>
+     *
```

Comment:
It is worth noting that if an IOException occurs, the reader and underlying stream are left in an unpredictable state?

With a normal or exceptional return should anything be said about the state of the underlying stream?
The corresponding method on Files ensures the reader is closed (and propagates to the underlying stream).

Reply:
Please see the changes in [4fac584](https://github.com/openjdk/jdk/pull/24728/commits/4fac584c5b19b3910e7a362656d2b406ad4ba189).